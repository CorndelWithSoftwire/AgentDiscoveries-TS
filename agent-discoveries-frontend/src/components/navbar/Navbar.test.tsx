import NavBar from './Navbar'
import { cleanup, render, screen } from '@testing-library/react'

afterEach(() => {
    cleanup()
})

describe('NavBar', () => {
  it('Matches the snapshot', () => {
    const navbar = render(<NavBar />)
    expect(navbar.asFragment()).toMatchSnapshot()
  })

  it('Must render on page', () => {
    render(<NavBar />)
    const navbar = screen.getByTestId('nav-bar')
    expect(navbar).toBeInTheDocument()
  })
})
