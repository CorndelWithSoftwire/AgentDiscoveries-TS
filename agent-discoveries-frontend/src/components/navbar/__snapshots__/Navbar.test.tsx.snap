// Vitest Snapshot v1

exports[`Navbar > Matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar"
    data-testid="navbar"
    id="navbar"
  >
    <h1
      class="title"
      id="navbar-title"
    >
      Agent Discoveries
    </h1>
    <div
      class="items"
      data-testid="navbar-items"
      id="navbar-items"
    >
      <a
        data-testid="navbar-item"
        href="/home"
      >
        Homepage
      </a>
      <a
        data-testid="navbar-item"
        href="/about"
      >
        About
      </a>
      <a
        data-testid="navbar-item"
        href="/profile"
      >
        Your Profile
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
